pkg/controller/aws/identity/iamrolepolicyattachment.go

```diff
@@ -0,0 +1,209 @@
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha1.IAMRolePolicyAttachmentGroupVersionKind),
+		resource.WithExternalConnecter(&rpaConnector{client: mgr.GetClient()}),
+		resource.WithManagedConnectionPublishers())
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha1.IAMRolePolicyAttachmentKindAPIVersion, v1alpha1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.IAMRolePolicyAttachment{}).
+		Complete(r)
+}
+
+type rpaConnector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) (iam.RolePolicyAttachmentClient, error)
+}
+
+func (conn *rpaConnector) Connect(ctx context.Context, mgd resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRolePolicyAttachment)
+	if !ok {
+		return nil, errors.New(errObjectNotIAMRolePolicyAttachment)
+	}
+
+	newClientFn := iam.NewRolePolicyAttachmentClient
+	if conn.newClientFn != nil {
+		newClientFn = conn.newClientFn
+	}
+
+	awsconfig, err := utils.RetrieveAwsConfigFromProvider(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create a new RolePolicyAttachmentClient")
+	}
+
+	return &rpaExternal{c}, nil
+}
+
+type rpaExternal struct {
+	client iam.RolePolicyAttachmentClient
+}
+
+func (e *rpaExternal) Observe(ctx context.Context, mgd resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRolePolicyAttachment)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errObjectNotIAMRolePolicyAttachment)
+	}
+
+	req := e.client.ListAttachedRolePoliciesRequest(&awsiam.ListAttachedRolePoliciesInput{
+		RoleName: aws.String(cr.Spec.RoleName),
+	})
+	req.SetContext(ctx)
+
+	observed, err := req.Send()
+	if err != nil {
+		return resource.ExternalObservation{}, errors.Wrapf(err, "Error getting IAMRolePolicyAttachments for role with name: %v", cr.Spec.RoleName)
+	}
+
+	var attachedPolicyObject *awsiam.AttachedPolicy
+	for i := 0; i < len(observed.AttachedPolicies); i++ {
+		if cr.Spec.PolicyArn == *observed.AttachedPolicies[i].PolicyArn {
+			attachedPolicyObject = &(observed.AttachedPolicies[i])
+			break
+		}
+	}
+
+	if attachedPolicyObject == nil {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.IAMRolePolicyAttachmentExternalState = cr.BuildExternalStateFromObservation(*attachedPolicyObject)
+
+	return resource.ExternalObservation{
+		ResourceExists: true,
+	}, nil
+}
+
+func (e *rpaExternal) Create(ctx context.Context, mgd resource.Managed) (resource.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRolePolicyAttachment)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errObjectNotIAMRolePolicyAttachment)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
```

Comment:
Should this really be Available? It seems like we could set the resource available, then fail to create it.

Reply:
good catch!